compiler-core/src/recipe.rs

```diff
@@ -0,0 +1,158 @@
+use std::fmt;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use crate::config::PackageConfig;
+use crate::{Error, Result};
+use hexpm::version::Range;
+use serde::de::{self, Deserializer, MapAccess, Visitor};
+use serde::ser::{Serialize, SerializeMap, Serializer};
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
+#[serde(untagged, remote = "Self")]
+pub enum Recipe {
+    Hex { version: Range },
+    Path { path: PathBuf },
+    Git { git: String },
+}
+
+impl Recipe {
+    pub fn version_range(&self) -> Result<Range, Error> {
+        match self {
+            Recipe::Hex { version: range } => Ok(range.clone()),
+            Recipe::Path { path } => {
+                let mut config_path = path.clone();
+                config_path.push("gleam.toml");
+                let toml = std::fs::read_to_string(&config_path).map_err(|_| {
+                    Error::DependencyResolutionFailed("Local dependency could not be found".into())
+                })?;
+                let config: PackageConfig = toml::from_str(&toml).map_err(|_| {
+                    Error::DependencyResolutionFailed(
+                        "Local dependency config could not be parsed".into(),
+                    )
+                })?;
```

Comment:
Let's use the existing error code for loading the config file. If we want a custom error `gleam.toml` being absent (which is a good idea!) let's check if the file exists first.